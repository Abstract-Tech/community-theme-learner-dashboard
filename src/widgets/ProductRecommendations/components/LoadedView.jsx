import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Container } from '@edx/paragon';
import { useIntl } from '@edx/frontend-platform/i18n';

import messages from '../messages';
import { courseShape, courseTypeToProductTypeMap } from '../utils';
import ProductCardContainer from './ProductCardContainer';

const MAX_OPEN_COURSE_RECOMMENDATIONS = 4;
const MIN_OPEN_COURSE_RECOMMENDATIONS = 2;

const LoadedView = ({ crossProductCourses, openCourses }) => {
  const { formatMessage } = useIntl();
  const includesCrossProductTypes = crossProductCourses.length === 2;

  const finalProductList = useMemo(() => {
    if (includesCrossProductTypes) {
      const openCourseList = openCourses ? openCourses.slice(0, MIN_OPEN_COURSE_RECOMMENDATIONS) : [];
      return crossProductCourses.concat(openCourseList);
    }
    return openCourses.slice(0, MAX_OPEN_COURSE_RECOMMENDATIONS);
  }, [crossProductCourses, openCourses, includesCrossProductTypes]);

  const courseTypes = [...new Set(finalProductList.map((item) => courseTypeToProductTypeMap[item.courseType]))];

  return (
    <div className="bg-light-200">
      <Container
        size="lg"
        className="recommendations-container pt-sm-5 pt-4.5 pb-2 pb-sm-4.5 bg-light-200"
      >
        <h2>
          {formatMessage(messages.recommendationsHeading)}
        </h2>
        <ProductCardContainer finalProductList={finalProductList} courseTypes={courseTypes} />
      </Container>
    </div>
  );
};

LoadedView.propTypes = {
  crossProductCourses: PropTypes.arrayOf(
    PropTypes.shape(courseShape),
  ).isRequired,
  openCourses: PropTypes.arrayOf(
    PropTypes.shape(courseShape),
  ).isRequired,
};

export default LoadedView;
