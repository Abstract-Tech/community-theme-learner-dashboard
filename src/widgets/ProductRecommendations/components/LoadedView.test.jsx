import React from 'react';
import { shallow } from 'enzyme';

import { mockCrossProductCourses, mockOpenCourses } from '../testData';
import LoadedView from './LoadedView';

describe('ProductRecommendations LoadedView', () => {
  it('matches snapshot', () => {
    expect(
      shallow(
        <LoadedView
          crossProductCourses={mockCrossProductCourses}
          openCourses={mockOpenCourses}
        />,
      ),
    ).toMatchSnapshot();
  });
});
