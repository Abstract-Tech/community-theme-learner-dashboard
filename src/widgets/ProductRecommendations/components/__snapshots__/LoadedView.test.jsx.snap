// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRecommendations LoadedView matches snapshot 1`] = `
<div
  className="bg-light-200"
>
  <Container
    className="recommendations-container pt-sm-5 pt-4.5 pb-2 pb-sm-4.5"
    size="lg"
  >
    <h2>
      You might also like
    </h2>
    <ProductCardContainer
      courseTypes={
        Array [
          "Executive Education",
          "Boot Camp",
          "Course",
        ]
      }
      finalProductList={
        Array [
          Object {
            "courseType": "executive-education-2u",
            "image": Object {
              "src": "https://www.image-2.com/ed79a49b-64c1-48d2-afdc-054bf921e38d-6a76ceb47dea.small.jpg",
            },
            "owners": Array [
              Object {
                "key": "HarvardX",
                "logoImageUrl": "http://www.image.com/ef72daf3-c9a1-4c00-ba37-b3514392bdcf-8839c516815a.png",
                "name": "Harvard University",
              },
            ],
            "prospectusPath": "course/introduction-to-computer-sceince",
            "title": "Introduction to Computer Science",
          },
          Object {
            "courseType": "bootcamp-2u",
            "image": Object {
              "src": "https://www.image-2.com/ed79a49b-64c1-48d2-afdc-054bf921e38d-6a76ceb47dea.small.jpg",
            },
            "owners": Array [
              Object {
                "key": "HarvardX",
                "logoImageUrl": "http://www.image.com/ef72daf3-c9a1-4c00-ba37-b3514392bdcf-8839c516815a.png",
                "name": "Harvard University",
              },
            ],
            "prospectusPath": "course/introduction-to-computer-sceince",
            "title": "Introduction to Computer Science",
          },
          Object {
            "courseType": "verified-audit",
            "image": Object {
              "src": "https://www.image-2.com/ed79a49b-64c1-48d2-afdc-054bf921e38d-6a76ceb47dea.small.jpg",
            },
            "owners": Array [
              Object {
                "key": "HarvardX",
                "logoImageUrl": "http://www.image.com/ef72daf3-c9a1-4c00-ba37-b3514392bdcf-8839c516815a.png",
                "name": "Harvard University",
              },
            ],
            "prospectusPath": "course/introduction-to-computer-sceince",
            "title": "Introduction to Computer Science",
          },
          Object {
            "courseType": "audit",
            "image": Object {
              "src": "https://www.image-2.com/ed79a49b-64c1-48d2-afdc-054bf921e38d-6a76ceb47dea.small.jpg",
            },
            "owners": Array [
              Object {
                "key": "HarvardX",
                "logoImageUrl": "http://www.image.com/ef72daf3-c9a1-4c00-ba37-b3514392bdcf-8839c516815a.png",
                "name": "Harvard University",
              },
            ],
            "prospectusPath": "course/introduction-to-computer-sceince",
            "title": "Introduction to Computer Science",
          },
        ]
      }
    />
  </Container>
</div>
`;
